common/persistence/versionHistory.go

```diff
@@ -49,11 +49,9 @@ func NewVersionHistoryItem(
 func NewVersionHistoryItemFromInternalType(
 	input *types.VersionHistoryItem,
 ) *VersionHistoryItem {
-
 	if input == nil {
-		panic("version history item is null")
+		return nil
```

Comment:
are callers assuming this function will always return non-nil or panic? If so we should go over them and handle nil case gracefully (potentially by returning a proper error) to avoid nil pointer exceptions

Reply:
I tried to do that in the previous PR, though it's certainly possible I missed a few locations. The value I expect to be nil on the first request and subsequently used to keep track of the workflow state. 

this case (CDNC-11702) I missed